src/hal/src/device.rs

```diff
@@ -396,6 +396,14 @@ pub trait Device<B: Backend>: fmt::Debug + Any + Send + Sync {
     /// [graphics pipelines][crate::pso::GraphicsPipelineDesc].
     unsafe fn create_shader_module(&self, spirv: &[u32]) -> Result<B::ShaderModule, ShaderError>;
 
+    /// Create a new shader module from the `naga` module.
+    unsafe fn create_shader_module_from_naga(
```

Comment:
Could Naga still be optional if we accepted an enum for `create_shader_module` (e.g. a `SpirV` variant and a feature-gated `Naga` variant)?

Reply:
I think Naga being optional is a bit orthogonal to whether we have an `enum` or not. In both cases we have a choice, the enum just requires `#[non_exhaustive]` on it. In fact, I tried to prototype it originally and then went back to a different entry point.
Now I don't have a strong preference one way or the other.